core/src/com/unciv/logic/city/CityConstructions.kt

```diff
@@ -225,7 +225,7 @@ class CityConstructions {
     fun turnsToConstruction(constructionName: String, useStoredProduction: Boolean = true): Int {
         val workLeft = getRemainingWork(constructionName, useStoredProduction)
         if (workLeft < 0) // we've done more work than actually necessary - possible if circumstances cause buildings to be cheaper later
-            return 1 // we'll finish this next turn
+            return 0 // we'll finish this next turn
 
```

Comment:
You can't finish a building mid-turn, even with boosts

Reply:
Suppose during a multiplayer game you are building something that has only 20 production remaining, and you have 10 production/turn in this specific city. Then it will of course say you have 2 turns remaining to build it. After pressing 'next turn', the production will be added, and while you are waiting for other players to do their turn, the city label notes that you have only 1 turn remaining. During your next turn you still have 1 turn remaining, but after pressing 'next turn' again, you have provided exactly enough work into the building for it to finish. In this the label would read 0 turns remaining. _However_, if you had 11 production/turn, at this point 'workLeft == -2', so instead (before this change), it would say you would still have 1 turn remaining for it to finish building. This change is only so that this minor inconsistency is fixed.

Reply:
Or, in other words: during multiplayer games it was sometimes possible for a construction to show '0 turns remaining' or '1 turn remaining' during another players turn, even though in both cases the construction would finish at the start of the next turn. This fixes this, but does introduce the more rare bug where if production requirements change for a building during your turn, it might say '0 turns remaining'.

Reply:
I see, thanks for the explanation!